src/agent.py

```diff
@@ -53,6 +62,65 @@ def __init__(self, charm: ops.CharmBase, state: State):
         charm.framework.observe(
             charm.on[AGENT_RELATION].relation_departed, self._on_agent_relation_departed
         )
+        # Event hooks for agent-discovery-ingress
+        charm.framework.observe(
+            ingress_observer.ingress.on.ready,
+            self.reconfigure_agent_discovery,
+        )
+        charm.framework.observe(
+            ingress_observer.ingress.on.revoked,
+            self.reconfigure_agent_discovery,
+        )
```

Comment:
The guidance is that event handlers should be called `_on_<event name>`, is that applicable here please?

Reply:
The reason why I didn't name it `_on_<event_name>` is that the logic of handlers for those two events are exactly the same, so I was hesitating to have a method that does nothing other than calling another method, and if I implement those 2 methods separately they will basically have duplicated code, so in the end I just used `reconfigure_agent_discovery` as handler. Do you we should still have `on_<event_name>` hook that calls the `reconfigure_agent_discovery` method?

Reply:
yes please 🙂

Reply:
got it, I added those 2 handlers.